packages/core/parcel-bundler/src/utils/ProxyHelper.js

```diff
@@ -0,0 +1,62 @@
+const httpProxy = require('http-proxy');
+const path = require('path');
+const fs = require('fs');
+
+/**
+ * Try to locate the package.json file of the app executing this.
+ * From that packaje.json reads an attribute 'proxy'.
+ *
+ * 'proxy' accepts a string or an object for configuring a proxy service.
+ *
+ * In case that a string is passed, will use it as a host to proxy
+ * and any request to '/api*' will be proxied.
+ *
+ * If an object is passed will lock for 2 keys:
+ * 'proxyHost': the host to proxy the requests
+ * 'proxyPath': path to match requests to be proxied.
+ *
+ * @returns A proxy object if the configuration is correct, null
+ *          otherwise.
+ */
+function ProxyHelper() {
+  // Try to locate first the package.json of the running app
+  const appDirectory = fs.realpathSync(process.cwd());
+  const pkg = path.resolve(appDirectory, 'package.json');
```

Comment:
I think you should use `Resolver.getPackager` here instead (an instance variable on the bundler in `Server.js` `middleware()`)

https://github.com/parcel-bundler/parcel/blob/6cbf2a65cdcd66af135ff9eca2a3799bde9f8a5b/packages/core/parcel-bundler/src/Asset.js#L131

Reply:
>  Another option is to configure the proxy on the `serve` function that is already async,  would you prefer to go that way?

That would mean that when you use Parcel's middleware via the API, the requests won't be proxied.
On the other hand, the current implementation using `process.cwd()` is certainly not correct, because the package file resolution algorithm used by the other code is: "go up in the folder structure until there is a package.json file", you'd want to do this for the entry point: `bundler.entryPoints[0]`.

The only solution I see currently is reimplementing this code to be synchronous (eventhough synchronous fs calls should really be avoided):
https://github.com/parcel-bundler/parcel/blob/c670decf1e8a039a80a47395bf916643d1955613/packages/core/parcel-bundler/src/Resolver.js#L422-L434

Reply:
What about a very simple approach, using `process.cwd()` instead of using the bundler. We try to find the package.json in your project defining how the proxy should work. 
A simple sync path check will help to understand if we should proceed reading the package.json file.

I understand that `process.cwd()` will produce different results depending on where do you execute your process, but do you think is a good tradeoff to ask people to start their dev server from their working directory?

Reply:
No, I launch mine from `src/server`